### ISSUE TITLE
TestLookupCNAME fails with lookup unknown: cannot extract nextServer from list

### ISSUE THREAD

**Issue by amery** [darvaza-proxy/resolver#21]
```
--- FAIL: TestLookupCNAME (0.60s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x66228d]

goroutine 7 [running]:
testing.tRunner.func1.2({0x68c620, 0x8d5830})
        /opt/golang/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
        /opt/golang/src/testing/testing.go:1399 +0x39f
panic({0x68c620, 0x8d5830})
        /opt/golang/src/runtime/panic.go:884 +0x212
darvaza.org/resolver.validateResp({0x0?, 0x7498b0?}, 0xc00001e110?, {0x747f60?, 0xc00005db40?})
        /home/amery/projects/licensly2/src/darvaza.org/resolver/tools.go:22 +0x2d
darvaza.org/resolver.LookupResolver.stepLookupCNAME({{0x747f00?, 0xc000010678?}}, {0x7498b0, 0xc00001e110}, {0xc0001fe7f0, 0xd})
        /home/amery/projects/licensly2/src/darvaza.org/resolver/resolver_cname.go:92 +0xbd
darvaza.org/resolver.LookupResolver.doLookupCNAME({{0x747f00?, 0xc000010678?}}, {0x7498b0, 0xc00001e110}, {0xc0001fe7f0, 0xd})
        /home/amery/projects/licensly2/src/darvaza.org/resolver/resolver_cname.go:54 +0x17e
darvaza.org/resolver.LookupResolver.LookupCNAME({{0x747f00?, 0xc000010678?}}, {0x7498b0, 0xc00001e110}, {0x6d04c0?, 0xf?})
        /home/amery/projects/licensly2/src/darvaza.org/resolver/resolver_cname.go:22 +0xac
darvaza.org/resolver.TestLookupCNAME(0xc0001681a0)
        /home/amery/projects/licensly2/src/darvaza.org/resolver/resolver_cname_test.go:24 +0x205
testing.tRunner(0xc0001681a0, 0x6fc3e8)
        /opt/golang/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
        /opt/golang/src/testing/testing.go:1493 +0x35f
FAIL    darvaza.org/resolver    2.235s
FAIL
make: *** [.tmp/gen.mk:26: test-root] Error 1
```

after #22 the crash is fixed, but when using RootLookuper (as supposed) it fails

```
$ make test
▶ generating index…
▶ test: root
--- FAIL: TestLookupCNAME (0.82s)
    resolver_cname_test.go:26: lookup unknown: cannot extract nextServer from list
FAIL
FAIL    darvaza.org/resolver    3.018s
FAIL
make: *** [.tmp/gen.mk:26: test-root] Error 1
```

**Comment by amery** [comment]
explosion is caused by Lookup() returning `"cannot extract nextServer from list"`

**Comment by amery** [comment]
```
$ make test
▶ generating index…
▶ test: root
--- FAIL: TestLookupCNAME (0.82s)
    resolver_cname_test.go:26: lookup unknown: cannot extract nextServer from list
FAIL
FAIL    darvaza.org/resolver    3.018s
FAIL
make: *** [.tmp/gen.mk:26: test-root] Error 1
```

**Comment by amery** [comment]
fixed by #25
